internal/parser/scanner/trie.go

```diff
@@ -0,0 +1,151 @@
+package scanner
```

Comment:
Why is this file on this branch? On here, we use slice iteration, we don't need a trie.

Reply:
WIll be removed. Its not attached to the main code.